test/helpers/utils.go

```diff
@@ -129,9 +129,6 @@ func WithTimeout(body func() bool, msg string, config *TimeoutConfig) error {
 	asyncBody := func(ch chan bool) {
 		success := body()
 		ch <- success
-		if success {
-			close(ch)
```

Comment:
When would we re-read from this channel? There is only ever one reader from the channel (line 144 below), and if there is success, we exit. 

I guess it's fine to not close it since GC will take care of it, but I don't understand the commit message given how the code is written.

Reply:
Yeah, I was on the fence. I decided to remove the close only because we "re-use" the channel. Still, I can't see how it would be re-read. I'll remove this part of the PR; I prefer to make the change small (since this PR won't change much, probably).